csrc/fusion_profiler.cpp

```diff
@@ -387,146 +443,156 @@ void FusionProfile::reset() {
   kernel_time_ms = 0.0;
 
-  input_bytes = 0;
-  output_bytes = 0;
-
   effective_bandwidth_gbs = 0.0;
   percentage_peak_bandwidth = 0.0;
 
+  input_bytes = 0;
+  output_bytes = 0;
+
   kernel_profiles.clear();
 }
 
-std::array<const char*, 25> column_strs{
-    "Fus#",      "NSegs",       "CuEvtTm(ms)",  "HstTm(ms)",
-    "CmpTm(ms)", "KerTm(ms)",   "EffBw(GB/s)",  "%PeakBw",
-    "S-Seg#",    "S-KerTm(ms)", "S-CmpTm(ms)",  "S-EffBw(GB/s)",
-    "S-%PeakBw", "S-In(MB)",    "S-Out(MB)",    "S-Smem[Dyn,Stat]",
-    "S-Regs",    "S-Grid",      "S-Block",      "S-Cluster",
-    "S-Dev",     "S-Stm",       "S-PkBw(GB/s)", "S-DeviceName",
-    "S-KerName"};
+const size_t FusionProfile::first_cupti_idx = 5;
+const std::vector<ProfileAttrDescriptor> FusionProfile::profile_attr_descs{
+    // column_header, verbose, segment, list, column_width, number,
+    // mantissa_width, unit_multiplier
+    {"Fus#", false, false, false, 5, true, 0, std::nullopt},
+    {"NSegs", false, false, false, 5, true, 0, std::nullopt},
+    {"CuEvtTm(ms)", false, false, false, 11, true, 3, std::nullopt},
+    {"HstTm(ms)", false, false, false, 9, true, 3, std::nullopt},
+    {"CmpTm(ms)", false, false, false, 9, true, 3, std::nullopt},
+    {"KerTm(ms)", false, false, false, 9, true, 3, std::nullopt},
+    {"EffBw(GB/s)", false, false, false, 11, true, 3, std::nullopt},
+    {"%PkBw", false, false, false, 7, true, 2, std::nullopt},
+    {"In(MB)", true, false, false, 8, true, 3, 1.0e-6},
+    {"Out(MB)", true, false, false, 9, true, 3, 1.0e-6},
+    {"S-Seg#", false, true, false, 6, true, 0, std::nullopt},
+    {"S-KerTm(ms)", false, true, false, 11, true, 3, std::nullopt},
+    {"S-CmpTm(ms)", true, true, false, 11, true, 3},
+    {"S-EffBw(GB/s)", false, true, false, 13, true, 3, std::nullopt},
+    {"S-%PkBw", false, true, false, 7, true, 2, std::nullopt},
+    {"S-In(MB)", false, true, false, 8, true, 3, 1.0e-6},
+    {"S-Out(MB)", false, true, false, 9, true, 3, 1.0e-6},
+    {"S-Smem[Dyn,Stat]", false, true, true, 16, false, 0, std::nullopt},
+    {"S-Regs", false, true, false, 6, true, 0, std::nullopt},
+    {"S-Grid", false, true, true, 16, true, 0, std::nullopt},
+    {"S-Block", false, true, true, 16, false, 0, std::nullopt},
+    {"S-Cluster", true, true, true, 16, false, 0, std::nullopt},
+    {"S-Dev", true, true, false, 5, true, 0, std::nullopt},
+    {"S-Stm", true, true, false, 5, true, 0, std::nullopt},
+    {"S-PkBw(GB/s)", true, true, false, 12, true, 3, std::nullopt},
+    {"S-DeviceName", true, true, false, 20, false, 0, std::nullopt},
+    {"S-KerName", false, true, false, 20, false, 0, std::nullopt}};
+
+namespace {
+// The operator* overloads are to satisfy the compiler and should not be called!
+double operator*(const std::basic_string<char>& a, double b) {
+  NVF_ERROR(false, "This types operator* overload should not be called!");
+  return 0.0;
+}
+
+template <typename T, size_t I>
+std::string to_string(const std::array<T, I>& cont) {
```

Comment:
Nit: `toString` following camelCase for C++ functions (ignore this if you are instead trying to keep consistent with a python snake_case interface).